src/resolvers/Mutation/createEvent.ts

```diff
@@ -174,7 +170,71 @@ export const createEvent: MutationResolvers["createEvent"] = async (
     }
   }
      */
+  if (!createdEvent) {
+    throw new Error(requestContext.translate("Failed to create event!")); // Adjust the error message accordingly
+  }
 
   // Returns the createdEvent.
-  return createdEvent.toObject();
+  return createdEvent[0].toObject();
 };
+
+export async function associateEventWithUser(
+  currentUser: any,
+  createdEvent: any
+) {
+  await EventAttendee.create({
+    userId: currentUser._id.toString(),
+    eventId: createdEvent._id,
+  });
+
+  await User.updateOne(
+    {
+      _id: currentUser._id,
+    },
+    {
+      $push: {
+        eventAdmin: createdEvent._id,
+        createdEvents: createdEvent._id,
+        registeredEvents: createdEvent._id,
+      },
+    }
+  );
+}
+
+export async function generateWeeklyRecurringInstances(
```

Comment:
Why are you using a loop here rather than writing query to directly handle such case. The choice is load on database or server? I would say database because loop here in server should always be avoided until there is no other way.